spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/trace/StackdriverTraceAutoConfiguration.java

```diff
@@ -221,8 +218,8 @@ public Sender stackdriverSender(GcpTraceProperties traceProperties,
 
 	@Bean
 	@ConditionalOnMissingBean
-	public Propagation.Factory stackdriverPropagation() {
-		return StackdriverTracePropagation.FACTORY;
+	public BaggagePropagation.FactoryBuilder baggagePropagationFactoryBuilder() {
+		return BaggagePropagation.newFactoryBuilder(StackdriverTracePropagation.FACTORY);
```

Comment:
right here will change with 0.17 we should use the correct brave.propagation package and add a comment that the underlying B3Propagation.Factory needs to be configured differently for sleuth 3.0

This can be a TODO I suppose as 3.0 isn't final anyway

Reply:
Do you mean that the `B3Propagation.Factory` would be configured differently in zipkin-gcp, because we're not configuring it here?

Reply:
I mean what we are overriding will be different configuration 2.2.x vs 3.0.x. Until we some annotation or otherwise to indicate the "real trace format" factory vs the wrapped one, this project will have to build differently for 2.2.x vs 3.0 or have some logic etc. @spencergibb mentioned maybe we can change 3.0 to use the `@Primary` annotation. Will think about it. Anyway, this is fine for 2.2.x